odoo/addons/base/models/res_company.py

```diff
@@ -114,15 +113,14 @@ def init(self):
         if hasattr(sup, 'init'):
             sup.init()
 
+    def _get_company_address_fields_matching(self):
+        """ Tools method returning a matching: company field name: partner field name """
+        return dict((field_name, field_name)
```

Comment:
Make it a regular list instead.
Will simplify the code a lot and we don't really need that complexity (yet).